mock-api/msw/db.ts

```diff
@@ -168,7 +205,10 @@ export const lookup = {
   vpcSubnet({ subnet: id, ...vpcSelector }: PP.VpcSubnet): Json<Api.VpcSubnet> {
     if (!id) throw notFoundErr('no subnet specified')
 
-    if (isUuid(id)) return lookupById(db.vpcSubnets, id)
+    if (isUuid(id)) {
+      ensureNoParentSelectors('subnet', vpcSelector)
+      return lookupById(db.vpcSubnets, id)
+    }
```

Comment:
There's a small downside to this implementation for double-nested resources like subnet: if you were to try to select a subnet by `(subnet name, VPC ID, project name)`, the thing that would blow up would be the VPC lookup, and the error would be "when VPC is specified by ID, project should not be specified." When I put it that way, it's not that confusing. But if you only look at the error message and not what endpoint the error is coming from, you can get thrown off.